src/util/PathUtil.ts

```diff
@@ -1,48 +1,28 @@
 import { posix, win32 } from 'path';
+import escapeStringRegexp from 'escape-string-regexp';
 import urljoin from 'url-join';
 import type { TargetExtractor } from '../http/input/identifier/TargetExtractor';
 import type { ResourceIdentifier } from '../http/representation/ResourceIdentifier';
 import type { HttpRequest } from '../server/HttpRequest';
 import { BadRequestHttpError } from './errors/BadRequestHttpError';
 
-// Characters to ignore when URL decoding the URI path to a file path.
-const pathComponentDelimiters = [ '/', '\\' ];
-// Regex to find all instances of encoded path component delimiters.
-const encodedDelimiterRegex = new RegExp(
-  `${pathComponentDelimiters.map((delimiter): string => encodeURIComponent(delimiter)).join('|')}`, 'giu',
+// URI path component delimiters and their escaped versions
+const pathComponentDelimiters = [ '/', '\\' ]
+  .flatMap((delimiter): string[] => [
+    delimiter,
+    encodeURIComponent(delimiter),
+    encodeURIComponent(delimiter).toLowerCase(),
+    encodeURIComponent(encodeURIComponent(delimiter)),
+    encodeURIComponent(encodeURIComponent(delimiter)).toLowerCase(),
+  ]);
+// Regex that matches URI path component delimiters and their escaped versions
+const pathComponentSplitter = new RegExp(
+  `(${pathComponentDelimiters.map(escapeStringRegexp).join('|')})`, 'gu',
+);
+// Map delimiters to their uppercase equivalent
+const normalizedDelimiters = Object.fromEntries(
+  pathComponentDelimiters.map((delimiter): [string, string] => [ delimiter, delimiter.toUpperCase() ]),
 );
```

Comment:
This made me think if perhaps we should handle canonical URLs differently? Currently if someone does a request to a URL that is not canonical we modify the URL internally and then use that URL from then on. E.g., someone does a GET to `/a&b` but internally we modify the URL to `/a%26b`, or in this case `/a%2fb` to `/a%2Fb`. Should we instead respond with a redirect to the canonical URL in case the input URL is not canonical? Probably not but maybe food for thought.

Reply:
I think it's okay; on an HTTP level they are seen as equivalent. (Not on an RDF level sadly.)